src/pages/Home.tsx

```diff
@@ -0,0 +1,114 @@
+import { useEffect, useRef, useState } from 'react';
+import { Outlet, useNavigate, useParams } from 'react-router';
+import { ITEMS_DISPLAY_LIMIT, searchProducts } from '../services/product';
+import { Product } from '../services/product.types';
+import ProductSearchInput from '../components/Product/Search/Input';
+import ProductSearchResult from '../components/Product/Search/Result';
+import { UNKNOWN_ERROR_MESSAGE } from '../constants/errorMessages';
+import Pagination from '../components/common/Pagination';
+import useSearchQuery from '../hooks/useSearchQuery';
+
+const HomePage = () => {
+  const navigate = useNavigate();
+  const navigateRef = useRef(navigate);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```

Comment:
in this case - it can be 
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```